### PR TITLE
Better test logging for local development and Travis runs

### PR DESCRIPTION
**Pull request by rsimha-amp**
- Replaces karma's `dots` reporter with `super-dots` on Travis. Now prints color coded check marks and crosses for passing / failing tests
- Replaces karma's `progress` reporter with `mocha` if `--testnames` or `--files` is passed in to the command line. If enabled, this will print the name of each test being run and its pass / fail status. Default behavior remains unchanged.
- Silences extremely verbose dependency install log messages for validator tests, since these currently account for more than half the size of a travis log. Errors will still be printed to the console.

#10482

### CODE REVIEW

**Comment by rsimha-amp** [comment]
/to @erwinmombay (overall), @honeybadgerdontcare (for validator changes)

**Comment by rsimha-amp** [comment]
@erwinmombay, @honeybadgerdontcare: Pinging again, since this is now ready for merge.

File: build-system/tasks/karma.conf.js [hunk 1]
```diff
@@ -44,7 +44,15 @@ module.exports = {
...
   },
 
-  reporters: [process.env.TRAVIS ? 'dots' : 'progress'],
+  reporters: [process.env.TRAVIS ? 'super-dots' : 'progress'],
+
+  superDotsReporter: {
+    icon: {
+      success : '✔',
+      failure : '✖',
+      ignore  : '?'
```
Comment (erwinmombay) [inline build-system/tasks/karma.conf.js]: add trailing ,

Comment (rsimha-amp) [inline build-system/tasks/karma.conf.js]: Done here and the line below.

**Comment by choumx** [comment]
@rsimha-amp How can you tell which tests are failing with this change? E.g. https://travis-ci.org/ampproject/amphtml/jobs/255041599

**Comment by rsimha-amp** [comment]
@choumx Hmm... I saw failure logs when I tested this out locally. Perhaps they're being silenced on Travis as a result of using `super-dots` instead of `dots`? I'll take a look right now, and if I can't see a cause, I'll revert back to `dots` on Travis.

Do you have a test that is known to fail? If not, I can try forcing a test to fail.

**Comment by rsimha-amp** [comment]
@choumx, I believe I've found the cause. Working on a fix.

**Pull request merged**